### ISSUE TITLE
[Question] Promisify

### ISSUE THREAD

**Issue by Sharlaan** [FoalTS/foal#893]
I noticed some simple internal functions like `isCommon` `verifyPassword` or `createJWT` implementation in Docs are forcedly promisified.

Any reason for that ?

**Comment by LoicPoullain** [comment]
Yes, the reason for that is not to stuck the event loop. The `isCommon` functions reads a file on the disk (where common passwords are listed) and `verifyPassword` calls the `pbkdf2` crypto utility. This operation might take time.

Let's imagine that the hard drive has a problem (for any reason) and it takes a big amount of time to read a file on it. With this async implementation, the server is still able to process other incoming requests while the file is being loaded.

**Comment by Sharlaan** [comment]
Ok Thanks for explanation :)
